Doc/library/itertools.rst

```diff
@@ -436,15 +436,25 @@ loops that truncate the stream.
           # islice('ABCDEFG', 2, None) --> C D E F G
           # islice('ABCDEFG', 0, None, 2) --> A C E G
           s = slice(*args)
-          it = iter(range(s.start or 0, s.stop or sys.maxsize, s.step or 1))
+          start, stop, step = s.start or 0, s.stop or sys.maxsize, s.step or 1
+          it = iter(range(start, stop, step))
           try:
               nexti = next(it)
           except StopIteration:
+              # Consume *iterable* up to the *start* position.
+              for i, element in zip(range(start), iterable):
+                  pass
               return
           for i, element in enumerate(iterable):
               if i == nexti:
                   yield element
-                  nexti = next(it)
+                  try:
+                      nexti = next(it)
+                  except StopIteration:
+                      # Consume to *stop*.
```

Comment:
For now, keep this consume-to-stop code as you currently have it.  I'm taking a while to mull over whether this needs to be part of the recipe or whether to count this distracting and exotic corner case as the "rough" part of the rough equivalent.